pygfx/renderers/wgpu/_wgpurenderer.py

```diff
@@ -212,78 +207,108 @@ def _record_render(self, command_encoder, q):
 
         render_pass.end_pass()
 
-    def get_render_list(self, scene: WorldObject, proj_screen_matrix: Matrix4):
-        """ Given a scene object, get a list of objects to render.
-        """
+    def _update_stdinfo_buffer(self, camera, physical_size, logical_size):
+        # Make sure we have a buffer object
+        if not hasattr(self, "_wgpu_stdinfo_buffer"):
+            self._wgpu_stdinfo_buffer = Buffer(
+                array_from_shadertype(stdinfo_uniform_type), usage="uniform"
+            )
+        # Update its data
+        stdinfo_data = self._wgpu_stdinfo_buffer.data
+        stdinfo_data["cam_transform"] = tuple(camera.matrix_world_inverse.elements)
+        stdinfo_data["projection_transform"] = tuple(camera.projection_matrix.elements)
+        stdinfo_data["physical_size"] = physical_size
+        stdinfo_data["logical_size"] = logical_size
+        # Upload to GPU
+        self._wgpu_stdinfo_buffer.update_range(0, 1)
+        self._update_buffer(self._wgpu_stdinfo_buffer)
 
-        # start by gathering everything that is visible and has a material
-        q = []
+    def get_render_list(self, scene: WorldObject):
+        """ Given a scene object, get a flat list of objects to render.
+        """
 
         def visit(wobject):
             nonlocal q
             if wobject.visible and hasattr(wobject, "material"):
                 q.append(wobject)
 
+        q = []
         scene.traverse(visit)
-
-        # next, sort them from back-to-front
-        def sort_func(wobject: WorldObject):
-            z = (
-                Vector3()
-                .set_from_matrix_position(wobject.matrix_world)
-                .apply_matrix4(proj_screen_matrix)
-                .z
-            )
```

Comment:
This sorting was here to try and minimize overdraw. Why did you decide to remove it?

Reply:
Because I think we should make it so that the order of rendering does not matter (order independent transparency). Blending happens per-fragment, not per-object (e.g. does not help for objects that intersect).

But we don't have that yet, so guess I'll turn it back on.

Reply:
* We don't have order independent transparency yet right?
* "Blending happens per-fragment" is kind of the point; if you render objects closest to the camera first, it's more likely that a portion of fragments can be skipped entirely thanks to the depth buffer being filled with closer objects sooner rather than later

Or am I missing the point here?

Reply:
My point is that ordering objects is not a guarantee that the rendering occurs correctly. The z-position of an object is e.g. its center of mass, or the center of its bounding box, but its vertices can be anywhere. So even though object A can be behind object B according to the sorting, there can still be many fragments of B that are in front of A.

Your point of performance is valid; you're doing sorting on the CPU to save work for the GPU. But it will depend heavily on the situation whether it will help or only load the CPU more (especially in Python ;))

Reply:
Something like a `render_order` flag or integer `level` (ThreeJS has something like that, right?) would therefore be better IMO. It gives the user a way to influence drawing order, but its much less costly (we can put the world-objects into bins instead of a single list when we collect them).

Reply:
Restored the sorting code. I don't see a noticeable difference for the collections example, so I guess the performance overhead is not that big.

Reply:
I'd like to discuss this more in our next video call :) thanks anyway